### PR TITLE
update SDL2

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: com.freerdp.FreeRDP.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "finish-args": [
         "--share=ipc",
         "--socket=wayland",
-        "--socket=fallback-x11",
+        "--socket=x11",
```
Comment (bbhtt) [inline com.freerdp.FreeRDP.json:22]: Why was this changed?

Comment (akallabeth) [reply com.freerdp.FreeRDP.json:22]: @bbhtt we´re shipping 3 RDP clients in the flatpak:
1. `sdl-freerdp` a `SDL2` based client, capable of running with `X11` or `Wayland`
2. `wlfreerdp` a wayland only client
3. `xfreerdp` a `X11` only client

while the `SDL` client works niche with `fallback-x11` (but also `x11` and `wayland`) `xfreerdp` fails if `fallback-x11` is used.

the default launched client is `sdl-freerdp` so this does not matter, but that one still lacks one or two features of the older `xfreerdp` that many want, so a common request is to be able to launch that one too.

with this change the current default works nice, but also the old `X11` client works without modifications. by just running `flatpak run --command=xfreerdp com.freerdp.FreeRDP` while that was not possible with the old permissions.
